code/modules/mod/mod_paint.dm

```diff
@@ -0,0 +1,181 @@
+#define MODPAINT_MAX_COLOR_VALUE 1.5
+#define MODPAINT_MIN_COLOR_VALUE 0
+#define MODPAINT_MAX_SECTION_COLORS 2
+#define MODPAINT_MIN_SECTION_COLORS 0.25
+#define MODPAINT_MAX_OVERALL_COLORS 4
+#define MODPAINT_MIN_OVERALL_COLORS 1.5
+
+/obj/item/mod/paint
+	name = "MOD paint kit"
+	desc = "This kit will repaint your MODsuit to something unique."
+	icon = 'icons/obj/clothing/modsuit/mod_construction.dmi'
+	icon_state = "paintkit"
+	var/obj/item/mod/control/editing_mod
+	var/atom/movable/screen/color_matrix_proxy_view/proxy_view
+	var/list/current_color
+
+/obj/item/mod/paint/Initialize(mapload)
+	. = ..()
+	current_color = color_matrix_identity()
+
+/obj/item/mod/paint/examine(mob/user)
+	. = ..()
+	. += span_notice("<b>Left-click</b> a MODsuit to change skin.")
+	. += span_notice("<b>Right-click</b> a MODsuit to recolor.")
+
+/obj/item/mod/paint/pre_attack(atom/attacked_atom, mob/living/user, params)
+	if(!istype(attacked_atom, /obj/item/mod/control))
+		return ..()
+	var/obj/item/mod/control/mod = attacked_atom
+	if(mod.active || mod.activating)
+		balloon_alert(user, "suit is active!")
+		return TRUE
+	var/secondary_attack = LAZYACCESS(params2list(params), RIGHT_CLICK)
```

Comment:
Split this into `pre_attack_secondary`--I intend to fix this part of the click chain at some point.

Reply:
Then move out the above checks into a proc or similar.